src/tracing_library.cpp

```diff
@@ -95,6 +96,24 @@ dd::Expected<dd::Tracer> TracingLibrary::make_tracer(
 #endif
 
   auto final_config = dd::finalize_config(config);
+
+  if constexpr (kNginx_flavor == nginx::flavor::ingress_nginx) {
+    // NOTE(@dmehala): ingress-nginx regularly poll an healthcheck endpoint.
+    // To avoid reporting traces, set the sampling rate to `0` for this
+    // endpoint. This is done after `finalize_config` because it can
+    // environment variables override the programmatic configuration.
```

Comment:
```suggestion
    // endpoint. This is done after `finalize_config` because it can
    // override programmatic configuration.
```

wth. got a stroke mid sentence 😂